### ISSUE TITLE
24-Bit Truecolor Support

### ISSUE THREAD

**Issue by AtomToast** [pimutils/khal#955]
I am trying to modify the colorscheme of khal and according to the urwid documentation it should be possible to specify any colors with a hex notation http://urwid.org/manual/displayattributes.html#bit-foreground-and-background-colors
However when I try to enter a hex value I get this error:

```
Traceback (most recent call last):
  File "/usr/bin/ikhal", line 5, in <module>
    main_ikhal()
  File "/usr/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python3.8/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/lib/python3.8/site-packages/khal/cli.py", line 479, in interactive_cli
    controllers.interactive(
  File "/usr/lib/python3.8/site-packages/khal/controllers.py", line 544, in interactive
    ui.start_pane(
  File "/usr/lib/python3.8/site-packages/khal/ui/__init__.py", line 1313, in start_pane
    loop = urwid.MainLoop(
  File "/usr/lib/python3.8/site-packages/urwid/main_loop.py", line 118, in __init__
    screen.register_palette(palette)
  File "/usr/lib/python3.8/site-packages/urwid/display_common.py", line 856, in register_palette
    self.register_palette_entry(*item)
  File "/usr/lib/python3.8/site-packages/urwid/display_common.py", line 923, in register_palette_entry
    basic = AttrSpec(foreground, background, 16)
  File "/usr/lib/python3.8/site-packages/urwid/display_common.py", line 540, in __init__
    raise AttrSpecError(('foreground/background (%s/%s) require ' +
urwid.display_common.AttrSpecError: foreground/background ('white'/'#383830') require more colors than have been specified (16).
```

So as far as I can tell khal has limited the colors to be accepted to the 16 standard colors?

I'd appreciate if this limitation could be removed

**Comment by mb720** [comment]
I just got a similar error (without having changed anything in the config) when starting `ikhal` 0.11.1 on Arch Linux:

    $ ikhal
    Traceback (most recent call last):
      File "/usr/bin/ikhal", line 5, in <module>
        main_ikhal()
      File "/usr/lib/python3.11/site-packages/click/core.py", line 1130, in __call__
        return self.main(*args, **kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/lib/python3.11/site-packages/click/core.py", line 1055, in main
        rv = self.invoke(ctx)
            ^^^^^^^^^^^^^^^^
      File "/usr/lib/python3.11/site-packages/click/core.py", line 1404, in invoke
        return ctx.invoke(self.callback, **ctx.params)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/lib/python3.11/site-packages/click/core.py", line 760, in invoke
        return __callback(*args, **kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/lib/python3.11/site-packages/click/decorators.py", line 26, in new_func
        return f(get_current_context(), *args, **kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/lib/python3.11/site-packages/khal/cli.py", line 501, in interactive_cli
        controllers.interactive(
      File "/usr/lib/python3.11/site-packages/khal/controllers.py", line 593, in interactive
        ui.start_pane(
      File "/usr/lib/python3.11/site-packages/khal/ui/__init__.py", line 1328, in start_pane
        loop = urwid.MainLoop(
              ^^^^^^^^^^^^^^^
      File "/usr/lib/python3.11/site-packages/urwid/event_loop/main_loop.py", line 127, in __init__
        screen.register_palette(palette)
      File "/usr/lib/python3.11/site-packages/urwid/display_common.py", line 989, in register_palette
        self.register_palette_entry(*item)
      File "/usr/lib/python3.11/site-packages/urwid/display_common.py", line 1070, in register_palette_entry
        mono = AttrSpec(mono, DEFAULT, 1)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/lib/python3.11/site-packages/urwid/display_common.py", line 554, in __init__
        raise AttrSpecError(
    urwid.display_common.AttrSpecError: foreground/background ('dark cyan'/'default') require more colors than have been specified (1).

**Comment by geier** [comment]
I believe this issue was fixed with #1252

@AtomToast if it persists, please re-open.

Also, if you are interested in coloring of ikhal, please see #1279
